### PR TITLE
HRCPP-203 Update build environment for 8.0.0 release

### PR DESCRIPTION
**Pull request by rigazilla**
The build process now checks minimal requirements for c++11 compilers and tests against infinispan 8.0.1 
server

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.2)
+        message(FATAL_ERROR "Clang version must be at least 3.2!")
+    endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MVSC")
+    # require at least MS Visual Studio  2013
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 18.0)
+        message(FATAL_ERROR "Microsoft Visual Studio version must be at least 2013!")
+    endif()
+else()
+    message(WARNING "You are using an unsupported compiler! Compilation has only been tested with Clang and GCC.")
```
Comment (tristantarrant) [inline CMakeLists.txt]: add MVSC to the warning

**Comment by rigazilla** [comment]
Added MVSC as supported compiler in the warning message

**Comment by rigazilla** [comment]
Fixed with the correct version for the infinispan server 8.0.1.Final

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 278](http://ci.infinispan.org/viewLog.html?buildId=32565&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:09:19

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 279](http://ci.infinispan.org/viewLog.html?buildId=32566&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:37

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 280](http://ci.infinispan.org/viewLog.html?buildId=32591&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:03

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 281](http://ci.infinispan.org/viewLog.html?buildId=32592&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:08

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 282](http://ci.infinispan.org/viewLog.html?buildId=32658&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:20

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 283](http://ci.infinispan.org/viewLog.html?buildId=32659&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:08

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 284](http://ci.infinispan.org/viewLog.html?buildId=32719&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:01

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 285](http://ci.infinispan.org/viewLog.html?buildId=32720&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:08

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 286](http://ci.infinispan.org/viewLog.html?buildId=32721&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:00:08

Process exited with code 1

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 287](http://ci.infinispan.org/viewLog.html?buildId=32722&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for c0ee92434d2f4165bd1e24ab67e81c531aca9e3d
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:59

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 288](http://ci.infinispan.org/viewLog.html?buildId=32762&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 4b758fe343f69a0579c1e29d4fe38975c54102d9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:52

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 289](http://ci.infinispan.org/viewLog.html?buildId=32763&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 4b758fe343f69a0579c1e29d4fe38975c54102d9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:09

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 290](http://ci.infinispan.org/viewLog.html?buildId=32767&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 4b758fe343f69a0579c1e29d4fe38975c54102d9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:09:24

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 291](http://ci.infinispan.org/viewLog.html?buildId=32768&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 4b758fe343f69a0579c1e29d4fe38975c54102d9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:54

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 292](http://ci.infinispan.org/viewLog.html?buildId=32794&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for f8e336b281588a18b8c8f02baffca124b5660785
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:02:41

Process exited with code 2

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 293](http://ci.infinispan.org/viewLog.html?buildId=32795&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for f8e336b281588a18b8c8f02baffca124b5660785
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:09

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 294](http://ci.infinispan.org/viewLog.html?buildId=32796&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for f8e336b281588a18b8c8f02baffca124b5660785
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:58

**Pull request merged**